test/PerpetualPositionRouter.t.sol

```diff
@@ -222,35 +218,42 @@ contract Fallback is PositionRouterTest {
 
   function testFork_FailedCallWhenExtraCalldataArgument() public {
     delegateApproval.approve(vethPositionRouterAddr, 1);
+
+    vm.expectRevert(PerpetualPositionRouter.InvalidCalldata.selector);
     (bool ok,) = payable(vethPositionRouterAddr).call(
-      abi.encodePacked(uint8(4), uint160(0), uint32(block.timestamp), uint96(0), uint96(100))
+      abi.encodePacked(
+        uint8(4), uint160(0), uint32(block.timestamp), uint96(10), uint96(100), uint96(0)
+      )
     );
-    assertFalse(ok);
+    assertTrue(ok);
```

Comment:
Contrary to what the forge book says, I don't think you can use `expectRevert` this way. I changed this to `vm.expectRevert(PerpetualPositionRouter.FunctionDoesNotExist.selector);` and the test still passes.

I think you'll need to delete the expectRevert, change `(book ok,)` to `(bool ok, bytes memory data)` and then `assertEq(data, PerpetualPositionRouter.InvalidCalldata.selector)`

Reply:
here's an example where the expect revert data is `error message`, but you'd replace that with the `PerpetualPositionRouter.InvalidCalldata.selecto`
```solidity
(bool ok, bytes memory data) = address(a).call(abi.encodeCall(A.foo, ()));
assertFalse(ok, "call should fail");

bytes memory expectedErr = abi.encodeWithSignature("Error(string)", ("error message"));
assertEq(data, expectedErr);
```

Reply:
Ref https://github.com/foundry-rs/foundry/issues/5159#issuecomment-1591995117

Reply:
nvm, ignore all of this haha. you're doing things correctly and the book is correct